packages/block-library/src/navigation-menu-item/edit.js

```diff
@@ -33,14 +34,41 @@ import {
 	BlockControls,
 	InnerBlocks,
 	InspectorControls,
-	URLPopover,
 	RichText,
+	__experimentalLinkControl as LinkControl,
 } from '@wordpress/block-editor';
-import {
-	Fragment,
-	useRef,
-	useState,
-} from '@wordpress/element';
+import { Fragment, useState, useEffect } from '@wordpress/element';
+
+/**
+ * It updates the link attribute when the
+ * link settings changes.
+ *
+ * @param {Function} setter Setter attribute function.
+ */
+const updateLinkSetting = ( setter ) => ( setting, value ) => {
+	if ( setting === 'new-tab' ) {
```

Comment:
This PR will become relevant here.

https://github.com/WordPress/gutenberg/pull/18285